aci_interface_policies.tf

```diff
@@ -434,3 +434,58 @@ module "aci_spine_fabric_interface_configuration" {
   shutdown     = each.value.shutdown
   role         = each.value.role
 }
+
+locals {
+  interface_state = flatten([
+    for node in local.nodes : [
+      for interface in try(node.interfaces, []) : {
+        key      = format("%s/%s/%s", node.id, try(interface.module, local.defaults.apic.interface_policies.nodes.interfaces.module), interface.port)
+        node_id  = node.id
+        module   = try(interface.module, local.defaults.apic.interface_policies.nodes.interfaces.module)
+        port     = interface.port
+        shutdown = try(interface.shutdown, local.defaults.apic.interface_policies.nodes.interfaces.shutdown)
+      }
+    ]
+    if(try(local.apic.auto_generate_switch_pod_profiles, local.defaults.apic.auto_generate_switch_pod_profiles) || try(local.apic.auto_generate_access_leaf_switch_interface_profiles, local.defaults.apic.auto_generate_access_leaf_switch_interface_profiles) || try(local.apic.auto_generate_access_spine_switch_interface_profiles, local.defaults.apic.auto_generate_access_spine_switch_interface_profiles)) && (length(var.managed_interface_policies_nodes) == 0 || contains(var.managed_interface_policies_nodes, node.id)) && try(local.apic.new_interface_configuration, local.defaults.apic.new_interface_configuration) == false
+  ])
+}
+
+module "aci_interface_state" {
+  source = "./modules/terraform-aci-interface-state"
+
+  for_each = { for int in local.interface_state : int.key => int if local.modules.aci_interface_state && var.manage_interface_policies }
+  node_id  = each.value.node_id
+  module   = each.value.module
+  port     = each.value.port
+  shutdown = each.value.shutdown
+}
+
+locals {
+  fex_interface_state = flatten([
+    for node in local.nodes : [
+      for fex in try(node.fexes, []) : [
+        for interface in try(fex.interfaces, []) : {
+          key      = format("%s/%s/%s/%s", node.id, fex.id, "1", interface.port)
+          node_id  = node.id
+          module   = fex.id
+          port     = interface.port
+          shutdown = try(interface.shutdown, local.defaults.apic.interface_policies.nodes.interfaces.shutdown)
+        }
+      ]
+    ] if(try(local.apic.auto_generate_switch_pod_profiles, local.defaults.apic.auto_generate_switch_pod_profiles) || try(local.apic.auto_generate_access_leaf_switch_interface_profiles, local.defaults.apic.auto_generate_access_leaf_switch_interface_profiles) || try(local.apic.auto_generate_access_spine_switch_interface_profiles, local.defaults.apic.auto_generate_access_spine_switch_interface_profiles)) && (length(var.managed_interface_policies_nodes) == 0 || contains(var.managed_interface_policies_nodes, node.id)) && try(local.apic.new_interface_configuration, local.defaults.apic.new_interface_configuration) == false
```

Comment:
You can remove the `(length(var.managed_interface_policies_nodes) == 0 || contains(var.managed_interface_policies_nodes, node.id))` code since that's already tested in the creation of the `local.nodes` variable in `main.tf`.

Reply:
Fixed this.